components/director/internal/domain/api/service_test.go

```diff
@@ -1 +1,264 @@
-package api
+package api_test
+
+import (
+	"context"
+	"fmt"
+	"testing"
+
+	"github.com/kyma-incubator/compass/components/director/internal/domain/api"
+	"github.com/kyma-incubator/compass/components/director/internal/domain/api/automock"
+	"github.com/kyma-incubator/compass/components/director/internal/model"
+	"github.com/kyma-incubator/compass/components/director/internal/tenant"
+	"github.com/pkg/errors"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/mock"
+	"github.com/stretchr/testify/require"
+)
+
+func TestService_Create(t *testing.T) {
+	// given
+	testErr := errors.New("Test error")
+
+	modelInput := model.APIDefinitionInput{
+		Name:          "Foo",
+		ApplicationID: "id",
+		TargetURL:     "https://test-url.com",
+		Spec:          &model.APISpecInput{},
+		Version:       &model.VersionInput{},
+	}
+
+	apiDefinitionModel := mock.MatchedBy(func(api *model.APIDefinition) bool {
```

Comment:
this won't be needed if id generation will be injected to the service as a dependency. BTW name of this variable is misleading, because it should be for me clear that this is matcher, not model itself